### PR TITLE
fix: notifications runtime permission

### PR DESCRIPTION
**Pull request by mrehan27**
### Changes

- Updated React Native SDK version to `1.0.1-alpha.2`
- Adds notification permission request for newer Android versions and iOS (using `react-native-permissions` library)
- Added support for locally testing React Native and Android SDK
- Updated Android target and compile SDK version to 13

### CODE REVIEW

File: components/IdentifyUser.js
Comment (ami-aman) [inline components/IdentifyUser.js:13]: This needs to be verified on iOS app as well. We have native code in place in iOS to seek user permissions for notifications and using react-native-permissions could possibly add a duplicate but I would like to add this possibility too in our docs to allow flexibility to customers to seek permissions anytime in their app and not just at the app launch !

**Review by ami-aman — APPROVED** [review]
LGTM, just one comment to test the app on iOS as well.

File: components/IdentifyUser.js
Comment (mrehan27) [reply components/IdentifyUser.js:13]: I agree as there could be some cases where customers might want to explain end users why they need to enable notifications before asking for the permission.

For testing on iOS, I think we should wait for #35 to be merged first, I'll then test this before merging.

**Pull request merged**
